src/rime/algo/syllabifier.cc

```diff
@@ -9,8 +9,11 @@
 #include <boost/range/adaptor/reversed.hpp>
 #include <rime/dict/prism.h>
 #include <rime/algo/syllabifier.h>
+#include <rime/gear/corrector.h>
```

Comment:
dependency graph of possible split modules:
```
core <-- {algo, dict} <-- {gear, lever}
```
`gear` depends on `algo`, what happens here is a circular dependency.
Is it easy to move `corrector` to `algo`?